autogluon/utils/tabular/ml/models/fastainn/tabular_nn_fastai.py

```diff
@@ -0,0 +1,339 @@
+import contextlib
+import logging
+import shutil
+import tempfile
+import time
+from functools import partial
+from pathlib import Path
+from typing import Any
+
+import numpy as np
+import pandas as pd
+from fastai.basic_data import DatasetType
+from fastai.basic_train import load_learner, Learner
+from fastai.callbacks import EarlyStoppingCallback, SaveModelCallback
+from fastai.data_block import FloatList
+from fastai.layers import LabelSmoothingCrossEntropy
+from fastai.metrics import root_mean_squared_error, mean_squared_error, mean_absolute_error, accuracy, FBeta, AUROC, Precision, Recall, r2_score
+from fastai.tabular import FillMissing, Categorify, Normalize, TabularList, tabular_learner
+from fastai.utils.mod_display import progress_disabled_ctx
+
+from .hyperparameters.parameters import get_param_baseline
+from .hyperparameters.searchspaces import get_default_searchspace
+from ..abstract.abstract_model import AbstractModel
+from ...constants import REGRESSION, BINARY, MULTICLASS
+from ....utils.loaders import load_pkl
+from ....utils.savers import save_pkl
+
+# FIXME: Has a leak somewhere, training additional models in a single python script will slow down training for each additional model. Gets very slow after 20+ models (10x+ slowdown)
+#  Slowdown does not appear to impact Mac OS
+# Reproduced with raw torch: https://github.com/pytorch/pytorch/issues/31867
+# https://forums.fast.ai/t/runtimeerror-received-0-items-of-ancdata/48935
+# https://github.com/pytorch/pytorch/issues/973
+# https://pytorch.org/docs/master/multiprocessing.html#file-system-file-system
+# Slowdown bug not experienced on Linux if 'torch.multiprocessing.set_sharing_strategy('file_system')' commented out
+# NOTE: If below line is commented out, Torch uses many file descriptors. If issues arise, increase ulimit through 'ulimit -n 2048' or larger. Default on Linux is 1024.
+# torch.multiprocessing.set_sharing_strategy('file_system')
+
+# MacOS issue: torchvision==0.7.0 + torch==1.6.0 can cause segfaults; use torch==1.2.0 torchvision==0.4.0
```

Comment:
I suspect that the maintainers will want you to lazy import and set a lot of these things, as we have for other uses of these libraries (mainly for torch).

While doing that you could also lazily evaluate if the versions meet your criteria here.

Reply:
Updated PR for lazy imports.